### PR TITLE
[Freddy CYR] iP

### PR DESCRIPTION
**Pull request by freddychenyouren2**
# SeeWhyAre Bot 👀 👀 👀 

> An Enhanced app of the skeletal Duke bot

Hello there! 👋 This is my chatbot improved from the Duke bot 🤖 that only has one greeting line for the users.

There are lots of improvements done to this bot 🔥 . Below are the main features that have been added:
* Add and Track your task completion. Available task types: Todo, Deadline, Event.
* View your current list of task
* Filter and Find your desired tasks based on their name
* Delete tasks from the list
* Save your task data to your hard disk and load them back the next time you use SeeWhyAre bot.

### Project Setup
Want to contribute to my fork? I am more than happy to hear (or read ehe) your suggestions! Here is how you can contribute to my fork:
1. Clone my repo using the command `git clone https://github.com/freddychenyouren2/ip.git` 
2. Open IntelliJ IDEA IDE (Please install if you haven't done so. You may download the Free Community Version)
3. Configure Java SDK setup under `File` -> `Project Structure`. This project uses ****Java 11**** . 
6. In the `build.gradle` file from the root directory, click the green Play button besides the `run` command. Alternatively press `Ctrl` twice and enter `gradlew run` which should run the app.
7. Hooray! You have started SeeWhyAre bot! Happy exploring!! 😄 

### By the way...
**I hope that you have done the following:**
- [X] Download my release and play with it.
- [ ] Have a look at my [personal website](https://freddychenyouren2.wixsite.com/freddy-cyr).
- [ ] Let's connect!! I am interested in new and interesting ideas to contribute to the society! Share with me your contacts (Telegram, Whatsapp, or LinkedIn) so that I may reach out to you :)

# Java Usage
SeeWhyAre bot uses Java programming langauge, which is primarily an Object-oriented langauge. If you are not familiar with Java, here is an example block of code for you:

``` xml {.line-numbers}
public class SeeWhyAre {
    public static void main(String[] args) {
        System.out.println("Are you Seeing Why now?");
    }
}
```

### CODE REVIEW

File: src/main/java/duke/util/Parser.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    public void parseInput(String userInput)
+            throws IOException, EmptyDescriptionException, InvalidCommandException {
+        try {
+            String firstWord = getCommand(userInput);
+            String[] words = userInput.split("\\s+"); // Split input by space, put into array
+            Command command; //Use enum
+            try {
+                command = Command.valueOf(firstWord);
+            } catch (IllegalArgumentException e) {
+                command = Command.UNKNOWN;
```
Comment (cxo05) [inline src/main/java/duke/util/Parser.java]: You can immediately throw the `InvalidCommandException` here and not need an `UNKNOWN` enum

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    mainClass.set("duke.Duke")
+}
+
+shadowJar {
+    archiveBaseName = "duke"
+    archiveClassifier = null
+    dependsOn("distZip", "distTar")
+}
+
+run{
```
Comment (cxo05) [inline build.gradle:60]: Good job on following the coding standard. This is the only styling issue I can find.

File: src/main/java/duke/DialogBox.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            fxmlLoader.setController(this);
+            fxmlLoader.setRoot(this);
+            fxmlLoader.load();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        //Need standardised photos to make better circle clips
+        dialog.setText(text);
+        //dialog.setMinHeight(Region.USE_PREF_SIZE);
```
Comment (RSXIX) [inline src/main/java/duke/DialogBox.java]: Maybe can remove these comments if you not using

File: src/main/java/duke/Duke.java [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+ * Individual Project
+ * SeeWhyAre Bot
+ * 31 Aug 2023
+ *
+ * @author Freddy Chen You Ren
+ */
+public class Duke {
+
+    // HORIZONTAL_LINE
+    public static String HORIZONTAL_LINE = "    ____________________________________________________________"; //60 underscores.
```
Comment (RSXIX) [inline src/main/java/duke/Duke.java]: good use of variables to reduce clutter in code

File: src/main/java/duke/Duke.java [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    public static boolean isFinished = false;
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    // Images
+    private final Image user = new Image(Objects.requireNonNull(this.getClass().getResourceAsStream("/images/ElonMusk.jpeg")));
```
Comment (RSXIX) [inline src/main/java/duke/Duke.java]: Maybe can add descriptor here so we know the type that is being referred to

File: src/main/java/duke/task/Event.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        String[] details = userInput.split("/from | /to");
+        //details[0] contains "deadline" plus duke.task description, need to erase "deadline". details[1] contains String deadline timing
+        StringBuilder message = new StringBuilder();
+
+        if (details.length == 3) {
+            String taskDescription = details[0].trim().replaceFirst("event", "").trim();
+            String startTime = details[1].trim();
+            String endTime = details[2].trim();
+
+            //Check if input date is valid.
```
Comment (RSXIX) [inline src/main/java/duke/task/Event.java:76]: good use of comments to make sure the code is readable

File: src/main/java/duke/util/Storage.java [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+        }
+    }
+
+    /**
+     * Checks whether a Task has already been done.
+     *
+     * @param task The task whose completion status is to be checked.
+     * @param completionStatus The completion status read from memory. 0 means not done, 1 means done.
+     */
+    protected void checkCompletionStatus(Task task, String completionStatus) {
```
Comment (RSXIX) [inline src/main/java/duke/util/Storage.java:200]: good use of protected

**Review by RSXIX — COMMENTED** [review]
Appreciated your use of comments but some of them felt hard to follow and I think you have alot of trial and error code left in the file which makes it feel cluttered. Coding standards seem to mostly be followed. Naming was mostly fine.
